### PR TITLE
custom question sets :)

### PR DESCRIPTION
**Pull request by josephmaxwellhyland**

### CODE REVIEW

File: back-end/db/question.ts [hunk 1]
```diff
@@ -51,8 +52,16 @@ export default {
             return [];
         }
     },
-    getQuestionsForQuiz: async (numQuestions: number): Promise<IQuestionnaireQuestion[]> => {
-        return await Question.aggregate().sample(numQuestions);
+    getQuestionsForQuiz: async (numQuestions: number, customMode: string): Promise<IQuestionnaireQuestion[]> => {
+        let questions: IQuestionnaireQuestion[] = [];
+        if (customMode) {
+            questions = await Question.find({ tags: customMode });
```
Comment (josephmaxwellhyland) [inline back-end/db/question.ts:58]: pick questions based on custom mode

File: back-end/interfaces/IGame.ts [hunk 2]
```diff
@@ -10,5 +10,6 @@ export default interface IGame {
 	playerQuestionnaires: PlayerQuestionnaire[],
 	quizQuestions: IQuizQuestion[],
 	currentQuestionIndex: number,
+	customMode: string,
```
Comment (josephmaxwellhyland) [inline back-end/interfaces/IGame.ts:13]: store `customMode` in the db for each game

File: back-end/interfaces/IQuestionnaireQuestion.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ export interface IQuestionnaireQuestion {
   id: string;
   text: string;
   quizText: string;
+  tags: string[];
```
Comment (josephmaxwellhyland) [inline back-end/interfaces/IQuestionnaireQuestion.ts:5]: store an array of `tags` on each question - used to determine which questions belong to which sets

File: front-end/src/extra/HostOpenButton.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+interface IHostOpenButtonProps {
+  symbol: string;
+  title: string;
+  description: string;
+  onClick?: () => any;
+  bgImage?: string;
+  disabled: boolean;
+}
+
+export default function HostOpenButton(props: IHostOpenButtonProps): React.ReactElement {
```
Comment (josephmaxwellhyland) [inline front-end/src/extra/HostOpenButton.tsx:12]: one component for the big buttons on the host landing page

File: front-end/src/host/HostApp.tsx [hunk 5]
```diff
@@ -190,7 +192,7 @@ export default function HostApp(props: IHostProps) {
 
     if (state === "lobby") {
       socket.emit("reload-players");
-      return <HostLobby socket={socket} gameId={gameId} />;
+      return <HostLobby socket={socket} gameId={gameId} classroomGame={customMode === "classroom"} />;
```
Comment (josephmaxwellhyland) [inline front-end/src/host/HostApp.tsx:195]: pass the custom mode down so it does not speak all names in classroom mode

File: front-end/src/host/HostOpen.tsx [hunk 6]
```diff
@@ -36,161 +128,10 @@ export default function HostOpen(props: IOpenProps) {
...
-              paddingLeft: "2vw",
-              paddingRight: "2vw",
-              textAlign: "left",
-            }}
-          >
-            Coming Soon. Pick a theme and get unique questions powered by
-            OpenAI!
-          </h1>
-        </div>
+        {showCustom ? customButtons : openButtons}
```
Comment (josephmaxwellhyland) [inline front-end/src/host/HostOpen.tsx:131]: refactor to show main screen _or_ custom buttons depending on current state

**Review by hylandtechoutreach — APPROVED** [review]
let's do it

**Pull request merged**